src/semigroup/object.js

```diff
@@ -1,11 +1,29 @@
 import { Semigroup } from '../semigroup';
 import propertiesOf from 'object.getownpropertydescriptors';
+import stable from '../stable';
 
-const { assign, getPrototypeOf } = Object;
+const { assign, getPrototypeOf, keys } = Object;
 
 Semigroup.instance(Object, {
   append(o1, o2) {
     let properties = assign({}, propertiesOf(o1), propertiesOf(o2));
-    return Object.create(getPrototypeOf(o1), properties);
+    return Object.create(getPrototypeOf(o1), cacheGetters(properties));
   }
-});
\ No newline at end of file
+});
+
+function cacheGetters(descriptors) {
```

Comment:
I don't know if this will introduce a circular dependency, but this would be cleaner as a `map` than a `reduce`:

```js
return map(desc => desc.get ? assign({}, desc, get: stable(desc.get) : desc, descriptors);
```

Reply:
If we do need to `reduce` though, might as well use `Foldable`

Reply:
Agree, but it causes `RangeError: Maximum call stack size exceeded`

Reply:
Here is what lodash does https://github.com/lodash/lodash/blob/master/map.js

I could use lodash map or just copy and paste that code. Thoughts?

Reply:
Does `map` or `foldl` infinite loop, or both?